portal/views/user.py

```diff
@@ -1887,3 +1888,73 @@ def set_table_preferences(user_id, table_name):
     db.session.commit()
 
     return jsonify(pref.as_json())
+
+
+@user_api.route('/user/<int:user_id>/messages')
+@oauth.require_oauth()
+@roles_required([ROLE.ADMIN, ROLE.STAFF, ROLE.INTERVENTION_STAFF])
+def get_user_messages(user_id):
+    """Returns simple JSON defining user email messages
+
+    Returns JSON of all messages where the receipient's email matches
+    the given user.
+    ---
+    tags:
+      - User
+    operationId: get_user_messages
+    parameters:
+      - name: user_id
+        in: path
+        description: TrueNTH user ID
+        required: true
+        type: integer
+        format: int64
+    produces:
+      - application/json
+    responses:
+      200:
+        description:
+          Returns JSON of the user's table view preferences.
+        schema:
+          id: user_messages
+          properties:
+            sender:
+              type: string
+              description: Email message sender
+            recipients:
+              type: string
+              description: Email message recipients
+            subject:
+              type: string
+              description: Email message subject
+            body:
+              type: string
+              description: Email message body
+            sent_at:
+              type: string
+              format: date-time
+              description: Datetime of when email message was sent
+            user_id:
+              type: integer
+              format: int64
+              description: TrueNTH user ID
+      401:
+        description:
+          if missing valid OAuth token or if the authorized user lacks
+          permission to view requested user_id
+    """
+    user = current_user()
+    if user.id != user_id:
+        current_user().check_role(permission='view', other_id=user_id)
+        user = get_user(user_id)
+    if user.deleted:
+        abort(400, "deleted user - operation not permitted")
+
+    messages = []
+
+    for em in EmailMessage.query.all():
```

Comment:
This looks rather inefficient.  Any reason we can't query directly for the matching set of messages?

Reply:
@pbugni I went with this route, because of a couple issues I bumped up with when testing:
* We want to look for `EmailMessages` where the `recipients` list contains the user email address exactly. But we can't just query where `recipients contains user.email` or `recipients like %user.email%, because that could get some inexact matches (for example, if `EmailMessages.recipients="blorp@gmail.com"` and `user.email="orp@gmail.com"`, we'd get a false match). So the matching there is fairly complex, if it's in-query
* `user.email` has different behavior in-query, than it does in straight variables. For example, say `user._email="__invite__blorp@gmail.com"`. When `user.email` is called directly (as I'm doing currently), we get `blorp@gmail.com` (which is what we want). However, when it's used in a sqlalchemy query, it's pulling the whole `__invite__blorp@gmail.com`. I mentioned this a few weeks ago in [this PT story comment thread](https://www.pivotaltracker.com/story/show/151111342). Something we should address at some point, but in the meantime, we need to be wary of using email as a param in sqlalchemy queries.

Reply:
@OptimusRhine fair enough.

Can you add a comment in the code, so we don't foolishly *optimize* those problems in